commons/src/main/java/org/archive/util/Base32.java

```diff
@@ -18,142 +18,25 @@
-                digit |= nextByte >> (8 - index);
-                i++;
-            } else {
-                digit = (currByte >> (8 - (index + 5))) & 0x1F;
-                index = (index + 5) % 8;
-                if (index == 0)
-                    i++;
-            }
-            base32.append(base32Chars.charAt(digit));
-        }
-
-        return base32.toString();
+      return BaseEncoding.base32().encode(bytes);
     }
-
     /**
-     * Decodes the given Base32 String to a raw byte array.
-     *
-     * @param base32
-     * @return Decoded <code>base32</code> String as a raw byte array.
-     */
-    static public byte[] decode(final String base32) {
-        int i, index, lookup, offset, digit;
-        byte[] bytes = new byte[base32.length() * 5 / 8];
-
-        for (i = 0, index = 0, offset = 0; i < base32.length(); i++) {
-            lookup = base32.charAt(i) - '0';
-
-            /* Skip chars outside the lookup table */
-            if (lookup < 0 || lookup >= base32Lookup.length) {
-                continue;
-            }
-
-            digit = base32Lookup[lookup];
-
-            /* If this digit is not in the table, ignore it */
-            if (digit == 0xFF) {
-                continue;
-            }
-
-            if (index <= 3) {
-                index = (index + 5) % 8;
-                if (index == 0) {
-                    bytes[offset] |= digit;
-                    offset++;
-                    if (offset >= bytes.length)
-                        break;
-                } else {
-                    bytes[offset] |= digit << (8 - index);
-                }
-            } else {
-                index = (index + 5) % 8;
-                bytes[offset] |= (digit >>> index);
-                offset++;
-
-                if (offset >= bytes.length) {
-                    break;
-                }
-                bytes[offset] |= digit << (8 - index);
-            }
-        }
-        return bytes;
-    }
-
-    /** For testing, take a command-line argument in Base32, decode, print in hex,
-     * encode, print
-     *
-     * @param args
+     * @deprecated Use {@link com.google.common.io.BaseEncoding#base32()}
      */
-    static public void main(String[] args) {
-        if (args.length == 0) {
-            System.out.println("Supply a Base32-encoded argument.");
-            return;
-        }
-        System.out.println(" Original: " + args[0]);
-        byte[] decoded = Base32.decode(args[0]);
-        System.out.print("      Hex: ");
-        for (int i = 0; i < decoded.length; i++) {
-            int b = decoded[i];
-            if (b < 0) {
-                b += 256;
-            }
-            System.out.print((Integer.toHexString(b + 256)).substring(1));
-        }
-        System.out.println();
-        System.out.println("Reencoded: " + Base32.encode(decoded));
+    @Deprecated
+    static public byte[] decode(final String base32)  {
+      return BaseEncoding.base32().decode(base32);
```

Comment:
This is throwing an exception on lowercase input such as in `ContentDigestHistoryTest.testBasics()`.

Reply:
Got it.